gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -157,13 +158,37 @@ def _read_variants(pipeline, known_args):
   return variants
 
 
-def _use_beam_header_merger(known_args):
+def _beam_merge_headers(known_args, pipeline_args):
   if known_args.representative_header_file:
-    return False
+    return
+
   match_results = FileSystems.match([known_args.input_pattern])
   if not match_results:
-    return False
-  return len(match_results[0].metadata_list) > _USE_HEADER_MERGER_THRESHOLD
+    return None
+
+  paths = [f.path for f in match_results[0].metadata_list]
+  if len(paths) > _USE_BEAM_READ_HEADER_THRESHOLD:
+    pipeline_options = GoogleCloudOptions(pipeline_args)
+
+    if pipeline_options.job_name:
+      pipeline_options.job_name += '-' + _MERGE_HEADERS_JOB_NAME
+    else:
+      pipeline_options.job_name = _MERGE_HEADERS_FILE_NAME
+
+    temp_directory = pipeline_options.temp_location or tempfile.mkdtemp()
+    known_args.representative_header_file = FileSystems.join(
+        temp_directory, _MERGE_HEADERS_FILE_NAME)
+
+    with beam.Pipeline(options=pipeline_options) as p:
+      headers = p
+      if len(paths) > _USE_BEAM_READ_ALL_HEADER_THRESHOLD:
```

Comment:
Consider just basing this on optimize_for_large inputs (see above for setting it if the threshold is higher than some value).

Reply:
Done.